boa/src/builtins/function/mod.rs

```diff
@@ -513,15 +513,27 @@ pub fn make_constructor_fn(
 ///     some other number of arguments.
 ///
 /// If no length is provided, the length will be set to 0.
-pub fn make_builtin_fn<N>(function: NativeFunctionData, name: N, parent: &Value, length: usize)
-where
+pub fn make_builtin_fn<N>(
+    function: NativeFunctionData,
+    name: N,
+    parent: &Value,
+    length: usize,
+    interpreter: &Interpreter,
+) where
     N: Into<String>,
 {
     let name = name.into();
     let _timer = BoaProfiler::global().start_event(&format!("make_builtin_fn: {}", &name), "init");
 
     // FIXME: function needs the Function prototype set.
     let mut function = Object::function(Function::builtin(Vec::new(), function), Value::null());
+    function.set_prototype(
+        interpreter
+            .global()
+            .get_field("Function")
+            .get_field("prototype"),
+    );
```

Comment:
We can remove the `set_prototype` call, instead we initialize it in the `Object::function()` since it takes the function and the prototype we can pass it instead of `Value::null()`

Also we can now remove the `FIXME` comment since we fixed the issue.

Reply:
Thanks for the review.